Tests/GRPCTests/GRPCReflectionServiceTests/GRPCReflectionServiceTests.swift

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright 2023, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import GRPC
+import GRPCReflectionService
+import NIOPosix
+import SwiftProtobuf
+import XCTest
+
+final class GRPCReflectionServiceTests: GRPCTestCase {
+  private var group: MultiThreadedEventLoopGroup?
+  private var server: Server?
+  private var channel: GRPCChannel?
+
+  private var fileDescriptorProto: Google_Protobuf_FileDescriptorProto? = nil
+  private var protos: [Google_Protobuf_FileDescriptorProto] = []
+  private var servicesNames: [String] = []
+
+  /// Creates the dependencies of the proto used in the testing context.
+  private func createDependencies() -> [String] {
+    var fileDependencies: [String] = []
+    for id in 1 ... 4 {
+      let inputMessage = Google_Protobuf_DescriptorProto.with {
+        $0.name = "inputMessage"
+        $0.field = [Google_Protobuf_FieldDescriptorProto.with {
+          $0.name = "inputField"
+          $0.type = .bool
+        }]
+      }
+
+      let outputMessage = Google_Protobuf_DescriptorProto.with {
+        $0.name = "inputMessage"
+        $0.field = [Google_Protobuf_FieldDescriptorProto.with {
+          $0.name = "outputField"
+          $0.type = .int32
+        }]
+      }
+
+      let method = Google_Protobuf_MethodDescriptorProto.with {
+        $0.name = "testMethod" + String(id)
+        $0.inputType = inputMessage.name
+        $0.outputType = outputMessage.name
+      }
+      let serviceDescriptor = Google_Protobuf_ServiceDescriptorProto.with {
+        $0.method = [method]
+        $0.name = "service" + String(id)
+      }
+      self.servicesNames.append("service" + String(id))
```

Comment:
We shouldn't be modifying our state here: it's not all obvious from the name of the function that it has side effects which can lead to bugs. For example your test could create the descriptors twice and end up with two copies a service in `serviceNames`.